tools/tpm2_clearcontrol.c

```diff
@@ -44,16 +44,63 @@ static bool clearcontrol(ESYS_CONTEXT *ectx) {
     return true;
 }
 
+static bool set_clearcontrol_auth_handle(const char *value) {
+
+    if (!strcmp(value, "p")) {
+        ctx.rh = ESYS_TR_RH_PLATFORM;
+        return true;
+    }
+
+    if (!strcmp(value, "l")) {
+        ctx.rh = ESYS_TR_RH_LOCKOUT;
+        return true;
+    }
+
+    LOG_ERR("Unknown or unsupported auth handle string."
+        " Specify p(latform)|l(ockout)");
+    return false;
+}
+
+static bool sanitize_input_arguments(void) {
+
+    if (!ctx.specifyhandle) {
+        ctx.rh = ESYS_TR_RH_LOCKOUT;
+    }
+
+    if (!ctx.set && !ctx.clear) {
+        LOG_ERR("Specify operation (set|clear)");
+        return false;
+    }
+
+    if (ctx.clear && ctx.rh == ESYS_TR_RH_LOCKOUT) {
+        LOG_ERR("Only platform hierarchy handle can be specified"
+            " for this operation");
+        return false;
+    }
+
+    return true;
+}
+
 static bool on_option(char key, char *value) {
 
+    bool result;
     switch (key) {
+    case 's':
+        ctx.disable_clear = 1;
+        ctx.set = true;
+        break;
```

Comment:
arguments, not options in the call to tpm2_options_new() set the on_arg handler and the parse the arguments like so:
 - argument count in argc must be equal to 1
 - the argument can be one of:
  - 0|c|clear
  - 1|s|set